File: Leetcode/916. Word Subsets.cpp
class Solution {
public:
    vector<string> wordSubsets(vector<string>& words1, vector<string>& words2) {
        vector<int> freq(26, 0);
        vector<int> t(26, 0);
        vector<string> res;
        for (string word : words2) {
            t.assign(26, 0);
            for (char i : word) t[i - 'a']++;
            for(int i=0;i<26;++i) freq[i]=max(freq[i],t[i]);
        }
        for(string word:words1){
            t.assign(26,0);
            for(char i:word){
                t[i-'a']++;
            }
            bool isUniversal=true;
            for(int i=0;i<26;++i){
                if(t[i]<freq[i]){
                    isUniversal=false; 
                    break;
                }
            }
            if(isUniversal) res.push_back(word);
        }
        return res;
    }
};

File: Leetcode/916. Word Subsets.py
class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        res=[]
        freq=[0]*26
        mf=[0]*26
        for word in words2:
            freq=[0]*26
            for c in word:
                freq[ord(c)-ord('a')]+=1
            for i in range(26):
                mf[i]=max(mf[i],freq[i])
        for word in words1:
            freq=[0]*26
            isUniversal=True
            for c in word:
                freq[ord(c)-ord('a')]+=1
            for i in range(26):
                if freq[i]<mf[i]:
                    isUniversal=False
                    break
            if isUniversal:
                res.append(word)
        return res
            
        
